### PR TITLE
Add support for relationships

### PR DESCRIPTION
**Pull request by sharplet**

### CODE REVIEW

File: Sources/ArgoJSONAPI/ArgoJSONAPI.swift [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
+public enum JSONAPI {
+}
 
 public func decode<T: JSONAPIDecodable>(_ object: Any) -> Decoded<T>
   where T.DecodedType == T
 {
   let json = JSON(object)
-
-  return (json <| "data") >>- Document<T>.decodeResource
+  return Document<T>.decodeResource(json)
```
Comment (gfontenot) [inline Sources/ArgoJSONAPI/ArgoJSONAPI.swift:11]: We should be able to remove the `Document<T>` here, right?

File: Sources/ArgoJSONAPI/ArgoJSONAPI.swift [hunk 2]
```diff
@@ -1,20 +1,20 @@
 import Argo
-import Runes
+
+public enum JSONAPI {
```
Comment (gfontenot) [inline Sources/ArgoJSONAPI/ArgoJSONAPI.swift]: What's the reasoning behind defining this here and then extending it elsewhere?

**Review by gfontenot — APPROVED** [review]
This is so awesome

File: Sources/ArgoJSONAPI/ArgoJSONAPI.swift (hunk 2 above)
Comment (sharplet) [reply Sources/ArgoJSONAPI/ArgoJSONAPI.swift]: It's just declaring a namespace basically, which has to happen _somewhere_, and so I chose to put it in the file named after the library. Do you think it's worth adding a doc comment to indicate that it's only intended as a namespace?

File: Sources/ArgoJSONAPI/ArgoJSONAPI.swift (hunk 1 above)
Comment (sharplet) [reply Sources/ArgoJSONAPI/ArgoJSONAPI.swift:11]: Nah, the return type is `Decoded<T>`, not `Document<T>`

Comment (gfontenot) [reply Sources/ArgoJSONAPI/ArgoJSONAPI.swift:11]: Ah, I see. Misread, my bad.

File: Sources/ArgoJSONAPI/ArgoJSONAPI.swift (hunk 2 above)
Comment (gfontenot) [reply Sources/ArgoJSONAPI/ArgoJSONAPI.swift]: Ah, gotcha. I think a public comment would probably be good, yeah.

**Comment by sharplet** [comment]
I think this is about done, but planning to wait for https://github.com/Quick/Quick/pull/711 to land before merging this.

**Pull request merged**